### ISSUE TITLE
Set KubeletDiskType to Temporary to be able to use all disks on azure vm:s

### ISSUE THREAD

**Issue by NissesSenap** [XenitAB/terraform-modules#554]
Azure got a preview feature of being able to set the kubelet disk type to temporary. This configures the kubelet to use the temporary disk that is available on all azure vm:s. This disk is bigger in general compared to many OS disk or ephemeral disks that is included in azure vm:s.

For example we can see that a `Standard_D4a_v4`  https://docs.microsoft.com/en-us/azure/virtual-machines/dav4-dasv4-series#dav4-series got 100 GiB of SSD temp storage.
It would be great to store all logs and container images on this disk.

https://github.com/Azure/azure-rest-api-specs/blob/5582a35deb1bfa4aa22bac8f1d51b7934ead94ac/specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-02-01/managedClusters.json#L3846-L3857

Sadly terraform currently don't support [kubelet_disk_type](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#kubelet_disk_type)

We should do some research and see for how long this feature have been in preview and depending on that it might be worth to try to implement it in the azurerm provider.

**Comment by NissesSenap** [comment]
Upstream issue created: https://github.com/hashicorp/terraform-provider-azurerm/issues/15449

**Comment by NissesSenap** [comment]
https://github.com/Azure/AKS/releases/tag/2022-04-03

```
Custom node config  for AKS is now generally available.
https://docs.microsoft.com/azure/aks/custom-node-configuration
```

**Comment by landerss1** [comment]
This attribute is available for the `default_node_pool` attribute in `azurerm_kubernetes_cluster` and in `azurerm_kubernetes_cluster_node_pool`.

**Comment by landerss1** [comment]
Solved by [#1152](https://github.com/XenitAB/terraform-modules/pull/1153).
